workbase/src/renderer/components/DataManagement/DataManagementUtils.js

```diff
@@ -59,9 +74,19 @@ async function loadRolePlayers(relationship, limitRolePlayers, limit, offset) {
   const promises = Array.from(roleplayers, async ([role, setOfThings]) => {
     const roleLabel = await role.label();
     await Promise.all(Array.from(setOfThings.values()).map(async (thing) => {
-      const type = await thing.type();
-      thing.type = await type.label();
-      thing.label = await buildLabel(thing);
+      switch (thing.baseType) {
+        case 'ENTITY':
+          await prepareEntity(thing);
+          break;
+        case 'ATTRIBUTE':
+          await prepareAttribute(thing);
+          break;
+        case 'RELATIONSHIP':
+          await prepareRelationship(thing);
+          break;
+        default:
+          // do nothing
```

Comment:
`throw "Unrecognised baseType of thing ["+thing.baseType+"]"` we should never get anything that is not entity/rel/attribute - if we ever do let's throw exception so that these things don't go unnoticed